src/source/Signaling/FileCache.c

```diff
@@ -0,0 +1,210 @@
+#define LOG_CLASS "SignalingFileCache"
+#include "../Include_i.h"
+
```

Comment:
Have a good description here of what the file structure will be, how many entries, how the cache entries will be updated. You can use a little ASCII picture to describe the entries.

Note that the size of the cache entries might be variable. The entries are delimited by \n
Note that the size of each field is variable and is delimited by ,

Most importantly, think whether you want to use a JSON format with JSON parser.

What I am worrying is that \n and , can be valid characters with the field entries which will then require complex escaping. With JSON I believe we have sanitized fields to not conflict with JSON format but needs to be checked.

Reply:
I dont think any of the fields can contain \n.

Reply:
what about ','? Validate it as this is cause havoc.

Reply:
channel name: [a-zA-Z0-9_.-]+
arn: arn:aws:kinesisvideo:[a-z0-9-]+:[0-9]+:[a-z]+/[a-zA-Z0-9_.-]+/[0-9]+
I dont think the end point will have either ',' or '\n'.